graph_theory/map_explorer.py

```diff
@@ -9,22 +9,42 @@ def __init__(self, map: list[list[int]]):
         self._island_count = 0
 
     def _check_row_and_cols_id(self, row_id: int, column_id: int) -> bool:
+        """
+        Check if the row_id and column_id are valid
+        :param row_id: row number
+        :param column_id: column number
+        :return: True if coordinates are valid
+                 False if coordinates are invalid
+
+        """
         if not (-1 < row_id < len(self._map) and -1 < column_id < len(self._map[row_id])):
             return False
         return True
 
     def _is_fragment_island(self, row_id: int, column_id: int) -> bool:
+        """
+        Check if given coordinates are invalid or visited,
+        if not calls itself to mark the entire island
+        :param row_id: row number
+        :param column_id: column number
+        :return: True if marking is completed
+                 False if coordinates are invalid, visited or there is no land
+        """
+        # 1. Check if coordinates are valid
         if not self._check_row_and_cols_id(row_id, column_id):
             return False
 
+        # 2. Check if coordinates were visited
         if (
             self._map[row_id][column_id] == self.MAP_NOT_ISLAND_MARK
             or self._visited_points[row_id][column_id]
         ):
             return False
 
+        # 3. Mark coordinates as visited
         self._visited_points[row_id][column_id] = True
 
+        # 4. Call itself to go through any land directly connected to the island
```

Comment:
Maybe that part behind should be extracted to another method, so this comment can be removed